File: Task-03/Python/subtask4.py
def daimond(n):
    pattern = []
    if(n%2!=0):
        for i in range(int((n+1)/2)):
            spaces= ' '*(n-i-1)
            stars='*'*(2*i+1)
            pattern.append(spaces+stars)
    
        for i in range(int((n-3)/2),-1,-1):
            spaces=' '*(n-i-1)
            stars='*'*(2*i+1)
            pattern.append(spaces+stars)
        return '\n'.join(pattern) 
    else:
        for i in range(int((n+1)/2)):
            spaces= ' '*(n-i-1)
            stars='*'*(2*i+1)
            pattern.append(spaces+stars)
    
        for i in range(int((n-1)/2),-1,-1):
            spaces=' '*(n-i-1)
            stars='*'*(2*i+1)
            pattern.append(spaces+stars)
    return '\n'.join(pattern)
    

inputfile='input.txt'
with open(inputfile,'r') as inputf:
   n = int(inputf.read().strip())
   
daimond_pattern=daimond(n)
   
outputfile='output.txt'
with open(outputfile,'w') as outputf:
    outputf.write(daimond_pattern)
                                                       
    
    

File: Task-03/C/subtask4.c
#include<stdio.h>
int main() {
int n;
FILE *file;
    file = fopen("input.txt", "r");
    fscanf(file, "%d", &n);
    fclose(file);
    file = fopen("output.txt", "w");
if(n%2!=0){
for(int i=1;i<=n;i=i+2){
for(int j=i;j<=n;j=j+2)
fprintf(file," ");
for(int k=1;k<=i;k++)
fprintf(file,"*");
fprintf(file,"\n");
}
for(int i=n-2;i>0;i=i-2){
for(int j=i;j<=n;j=j+2)
fprintf(file," ");
for(int k=1;k<=i;k++)
fprintf(file,"*");
fprintf(file,"\n");
}
}
else{
for(int i=1;i<=n;i=i+2){
for(int j=i;j<=n;j=j+2)
fprintf(file," ");
for(int k=1;k<=i;k++)
fprintf(file,"*");
fprintf(file,"\n");
}
for(int i=n;i>0;i=i-2){
for(int j=i;j<=n;j=j+2)
fprintf(file," ");
for(int k=1;k<i;k++)
fprintf(file,"*");
fprintf(file,"\n");
}
}
fclose(file);
printf("Pattern written to output.txt .\n");
return 0;
}
